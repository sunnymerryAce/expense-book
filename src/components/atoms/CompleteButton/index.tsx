import React from "react";
import Fab from "@material-ui/core/Fab";
import DoneIcon from "@material-ui/icons/Done";
interface CompleteButtonProps {
  handleClick: (props: unknown) => unknown;
}

const CompleteButton: React.FC<CompleteButtonProps> = (props) => {
  return (
    <Fab onClick={props.handleClick} color="secondary">
      <DoneIcon />
    </Fab>
  );
};

export default CompleteButton;
