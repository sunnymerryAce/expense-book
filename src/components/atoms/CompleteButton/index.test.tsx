import React from "react";
import CompleteButton from "./";
import { shallow } from "enzyme";

describe("<CompleteButton />", () => {
  let container: any = null;
  const handleClickFunction = jest.fn();

  beforeEach(() => {
    container = shallow(<CompleteButton handleClick={handleClickFunction} />);
  });

  afterEach(() => {
    container && container.unmount();
    container = null;
  });

  test("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot();
  });

  test("should have proper props", () => {
    expect(container.find(".CompleteButton").props()).toMatchObject({
      onClick: handleClickFunction,
    });
  });

  test("click event handler should be triggered", () => {
    container.simulate("click");
    expect(handleClickFunction).toBeCalled();
  });
});
