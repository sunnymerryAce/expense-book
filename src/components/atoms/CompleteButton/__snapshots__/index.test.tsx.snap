// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompleteButton /> should match the snapshot 1`] = `"<button class=\\"MuiButtonBase-root MuiFab-root CompleteButton MuiFab-secondary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiFab-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z\\"></path></svg></span></button>"`;
